src/main/scala/com/cognite/spark/datasource/SdkV1Relation.scala

```diff
@@ -75,6 +85,16 @@ abstract class SdkV1Relation[A, T <: Readable[A, IO], C: Decoder](
       ()
     })
 
+  def toRow(item: A, requiredColumns: Array[String]): Row =
+    if (requiredColumns.isEmpty) {
+      toRow(item)
+    } else {
+      val fieldNamesInOrder = item.getClass.getDeclaredFields.map(_.getName)
+      val indices = requiredColumns.map(f => fieldNamesInOrder.indexOf[String](f))
+      val fullRow = toRow(item)
+      Row.fromSeq(indices.map(fullRow.get))
```

Comment:
Maybe `idx => fullRow.get(idx)` would be easier to follow (at least for me - I had to double check the type of `fullRow`)

Reply:
Agreed, tried to make it a bit clearer now.